win32ss/gdi/ntgdi/path.c

```diff
@@ -1719,93 +1720,30 @@ PATH_StrokePath(
 
 #define round(x) ((int)((x)>0?(x)+0.5:(x)-0.5))
 
-static
-PPATH
-FASTCALL
-PATH_WidenPath(DC *dc)
+PPATH FASTCALL
+IntGdiWidenPath(PPATH pPath, UINT penWidth, UINT penStyle, FLOAT eMiterLimit)
 {
-    INT i, j, numStrokes, numOldStrokes, penWidth, penWidthIn, penWidthOut, size, penStyle;
-    PPATH pPath, flat_path, pNewPath, *pStrokes = NULL, *pOldStrokes, pUpPath, pDownPath;
-    EXTLOGPEN *elp;
+    INT i, j, numStrokes, numOldStrokes, penWidthIn, penWidthOut;
+    PPATH flat_path, pNewPath, *pStrokes = NULL, *pOldStrokes, pUpPath, pDownPath;
     BYTE *type;
-    DWORD obj_type, joint, endcap, penType;
-    PDC_ATTR pdcattr = dc->pdcattr;
-
-    pPath = PATH_LockPath(dc->dclevel.hPath);
-    if (!pPath)
-    {
-        EngSetLastError( ERROR_CAN_NOT_COMPLETE );
-        return NULL;
-    }
-
-    if (pPath->state != PATH_Closed)
-    {
-        DPRINT("PWP 1\n");
-        PATH_UnlockPath(pPath);
-        EngSetLastError(ERROR_CAN_NOT_COMPLETE);
-        return NULL;
-    }
-
-    size = GreGetObject(pdcattr->hpen, 0, NULL);
-    if (!size)
-    {
-        DPRINT("PWP 2\n");
-        PATH_UnlockPath(pPath);
-        EngSetLastError(ERROR_CAN_NOT_COMPLETE);
-        return FALSE;
-    }
-
-    elp = ExAllocatePoolWithTag(PagedPool, size, TAG_PATH);
-    if (elp == NULL)
-    {
-        DPRINT("PWP 3\n");
-        PATH_UnlockPath(pPath);
-        EngSetLastError(ERROR_OUTOFMEMORY);
-        return FALSE;
-    }
-
-    GreGetObject(pdcattr->hpen, size, elp);
-
-    obj_type = GDI_HANDLE_GET_TYPE(pdcattr->hpen);
-    if (obj_type == GDI_OBJECT_TYPE_PEN)
-    {
-        penStyle = ((LOGPEN*)elp)->lopnStyle;
-    }
-    else if (obj_type == GDI_OBJECT_TYPE_EXTPEN)
-    {
-        penStyle = elp->elpPenStyle;
-    }
-    else
-    {
-        DPRINT("PWP 4\n");
-        EngSetLastError(ERROR_CAN_NOT_COMPLETE);
-        ExFreePoolWithTag(elp, TAG_PATH);
-        PATH_UnlockPath(pPath);
-        return FALSE;
-    }
-
-    penWidth = elp->elpWidth;
-    ExFreePoolWithTag(elp, TAG_PATH);
+    DWORD joint, endcap, penType;
 
     endcap = (PS_ENDCAP_MASK & penStyle);
     joint = (PS_JOIN_MASK & penStyle);
     penType = (PS_TYPE_MASK & penStyle);
 
     /* The function cannot apply to cosmetic pens */
-    if (obj_type == GDI_OBJECT_TYPE_EXTPEN && penType == PS_COSMETIC)
+    if (penType == PS_COSMETIC)
```

Comment:
Before this checked also that the type was not {{GDI_OBJECT_TYPE_PEN}}.
This check was wrong?

Reply:
I added a check in PATH_WidenPath. IntGdiWidenPath doesn't need this check. Thanks.